### PR TITLE
add passive mode for BMP

### PR DESCRIPTION
**Pull request by tobikris**
Hey, I had a usecase for a passive BMP connection (e.g. the router should initiate the TCP connection).

I tried to add it to the best of my understanding of the inner workings of bio-rd.
I am not sure how to add a unit test for it though.

Side-note: one of the existing test cases is failing even on master (TestFSMAFIInitDispose).

### CODE REVIEW

File: protocols/bgp/server/bmp_server.go [hunk 1]
```diff
@@ -52,15 +53,116 @@ func NewServer(keepalivePeriod time.Duration) *BMPServer {
...
+		}
+		go b.handleConnection(c)
+	}
+}
+
+func (b *BMPServer) handleConnection(c net.Conn) {
+	remoteIP := c.RemoteAddr().(*net.TCPAddr).IP
+
+	var r *Router
+	for _, router := range b.getRouters() {
```
Comment (taktv6) [inline protocols/bgp/server/bmp_server.go:94]: Let's use the GetRouter() method here. Then just check for passive flag.

File: protocols/bgp/server/bmp_server.go [hunk 2]
```diff
@@ -52,15 +53,116 @@ func NewServer(keepalivePeriod time.Duration) *BMPServer {
...
+			"router":         r.Name(),
+		}).Error("router already connected, dropping connection")
+		c.Close()
+		return
+	}
+
+	b.establishConnection(c, r, true)
+}
+
+func (b *BMPServer) establishConnection(c net.Conn, r *Router, passive bool) error {
```
Comment (taktv6) [inline protocols/bgp/server/bmp_server.go:121]: Not happy with this naming. Let's rename establishConnection to handleConnection. The current handleConnection should be validateConnection() bool. Move the call of establishConnection into the Listen() method and call it only if validateConnection() returned true.

File: protocols/bgp/server/bmp_server.go (hunk 1 above)
Comment (tobikris) [reply protocols/bgp/server/bmp_server.go]: Thanks, changed it.

File: protocols/bgp/server/bmp_server.go (hunk 2 above)
Comment (tobikris) [reply protocols/bgp/server/bmp_server.go]: I changed the names and moved the call.

**Review by taktv6 — APPROVED** [review]

**Pull request merged**